src/emoji-picker/components/EmojiPicker.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+/* eslint-disable react/require-default-props */
+import React, { useState, useEffect, useCallback } from 'react';
+import Picker from '@emoji-mart/react';
+import { connect } from 'react-redux';
+import { Actions, ActionPayload } from '@twilio/flex-webchat-ui';
+
+import { AseloWebchatState } from '../../aselo-webchat-state';
+import { toggleEmojiPicker } from '../emoji-state';
+import { Popup } from './emoji-styles';
+import { blockedEmoji } from './blockedEmojis';
+import { getLocale } from './localizedEmojiPicker';
+
+const EmojiPicker = ({ channelSid, isPickerOpen, onToggleEmojiPicker, language }: Props) => {
+  const [inputText, setInputText] = useState('');
+
+  type onEmojiSelectPayload = {
+    native: string;
+  };
+
+  const concatEmoji = (input: string, emoji: string) => {
+    if (input.length === 0) {
+      return emoji;
+    }
+    return `${inputText}${emoji}`;
+  };
+
+  const handleSelectEmoji = useCallback(
+    (payload: onEmojiSelectPayload) => {
+      const body = concatEmoji(inputText, payload.native);
+      Actions.invokeAction('SetInputText', {
+        channelSid,
+        body,
+      });
+
+      onToggleEmojiPicker();
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [inputText, channelSid],
+  );
+
+  /**
+   * Adds two listeners: Action.SetInputText to track the inputText value
+   * Action.SendMessage to track and set inputText to an empty string
+   */
+  useEffect(() => {
+    const inputTextListener = (event: ActionPayload) => {
+      setInputText(event.body);
+    };
+    Actions.addListener('afterSetInputText', inputTextListener);
+
+    Actions.addListener('afterSendMessage', () => {
+      setInputText('');
+    });
+
+    return () => {
+      Actions.removeListener('afterSetInputText', inputTextListener);
+      Actions.addListener('afterSendMessage', () => {
+        setInputText('');
+      });
+    };
```

Comment:
on line 73 I believe you were supposed to `removeListener` instead of adding. For doing that you'll probably need to extract the callback `() => { setInputText('') }` to constant. Just look at line 72 for reference.